build.gradle

```diff
@@ -24,19 +9,19 @@ repositories {
     mavenCentral()
 }
 
-dependencies {
-    if (jcHomeSet) {
-        testImplementation(group: 'com.klinec', name: 'jcardsim', version: '3.0.5.11') {
-            // Javacard will be provided by the user at runtime through the JC_HOME env var
-            exclude(group: 'oracle.javacard', module: 'api_classic')
-        }
-    } else {
-        // Perform a full-test build, since there's no javacard SDK
-        implementation(group: 'com.klinec', name: 'jcardsim', version: '3.0.5.11')
-    }
-    testImplementation group: 'org.junit.jupiter', name: 'junit-jupiter-api', version: '5.8.1'
+// Create a new configuration for JCDK compilation
+configurations {
+    jcdk
+}
 
-    testRuntimeOnly group: 'org.junit.jupiter', name: 'junit-jupiter-engine', version: '5.8.1'
+dependencies {
+    // JCDK dependencies for CAP generation
+    jcdk files('../jckit/lib/api_classic-3.0.5.jar')
+    
+    // Main and test dependencies use JCardSim
+    implementation files('../jcardsim-3.0.5.jar')
```

Comment:
JCardSim shouldn't be a dependency for the javacard applet. It's only used for testing builds.